rust/agents/validator/src/submit.rs

```diff
@@ -182,7 +160,13 @@ impl ValidatorSubmitter {
         while tree.count() as u32 <= correctness_checkpoint.index {
             if let Some(insertion) = self
                 .message_db
-                .retrieve_merkle_tree_insertion_by_leaf_index(&(tree.count() as u32))?
+                .retrieve_merkle_tree_insertion_by_leaf_index(&(tree.count() as u32))
+                .unwrap_or_else(|_| {
+                    panic!(
+                        "Database is missing merkle insertion for index {}",
```

Comment:
nit: iiuc this returns a Result<Option<_>> where an Err is just a db error but Ok(None) indicates that the db is missing the merkle insertion for that index, which is expected in this code path - should the panic message indicate an error occurred (which is not expected) and share the err?